File: Chapter 02/Python/section_2_3_2_maintainability/cart_item.py
"""
This module defines the CartItem class.
"""

class CartItem:
    """
    Represents a single item within a shopping cart.
    
    This class is a simple data structure, often called a data class. 
    Its primary role is to hold data about a cart item, not to contain complex business logic.
    """
    def __init__(self, name, price):
        """
        Initializes a new CartItem.
        
        Args:
            name (str): The name of the product.
            price (float): The price of a single unit of the product.
        """
        self.name = name
        self.price = price

File: Chapter 02/C#/Section_2_3_2_Maintainability/CartItem.cs
namespace Chapter02.ShoppingCartExample;

/// <summary>
/// Represents a single item within a shopping cart.
/// This class is a simple data structure (also known as a Plain Old CLR Object or POCO).
/// Its only job is to hold data about a cart item, not to perform any business logic.
/// </summary>
public class CartItem
{
    /// <summary>
    /// The name of the product.
    /// </summary>
    public required string Name { get; set; }

    /// <summary>
    /// The price of a single unit of the product.
    /// </summary>
    public decimal Price { get; set; }
}